fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -61,4 +64,18 @@ default boolean allowNbtUpdateAnimation(PlayerEntity player, Hand hand, ItemStac
 	default boolean allowContinuingBlockBreaking(PlayerEntity player, ItemStack oldStack, ItemStack newStack) {
 		return false;
 	}
+
+	/**
+	 * Return the attribute modifiers to apply when this stack is worn in a living entity equipment slot.
+	 * Stack-aware version of {@link Item#getAttributeModifiers(EquipmentSlot)}.
+	 *
+	 * <p>Note that attribute modifiers are only updated when the stack changes, i.e. when {@code ItemStack.areEqual(old, new)} is false.
+	 *
+	 * @param stack the current stack
+	 * @param slot  the equipment slot this stack is in
+	 * @return the attribute modifiers
+	 */
+	default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers(ItemStack stack, EquipmentSlot slot) {
```

Comment:
Should we pre/suffix this with fabric to prevent possible conflicts with other mappings/mods?

Reply:
wouldn't that mean that the api is redundant anyway? there should be no vanilla method with the same signature and purpose

Reply:
> @AlphaMode do you have a list of "reasonable" methods to provide?
> 
> `isSuitableFor` for sure since it is also needed to properly replace the old tool api, but the others don't seem as useful?

Yeah the other ones aren't actually a part of the item actual item class.
I would just look around in the item class to add methods that don't pass the ItemStack and make a new variant that does.
Like onItemEntityDestroyed or isDamageable could have a variant that passes the ItemStack

Reply:
> Like onItemEntityDestroyed or isDamageable could have a variant that passes the ItemStack

Bad examples: `onItemEntityDestroyed` already has the ItemStack since it receives the `ItemEntity`, and `isDamageable` requires too much patching imo. :P

I will only add `getAttributeModifiers` and `isSuitableFor` in this PR, for the others we will see later.